udata_hydra/routes/resources.py

```diff
@@ -1,20 +1,25 @@
 import json
+import uuid
 
 from aiohttp import web
 from asyncpg import Record
 from marshmallow import ValidationError
 
 from udata_hydra.db.resource import Resource
-from udata_hydra.schemas import ResourceSchema
-from udata_hydra.utils import get_request_params
+from udata_hydra.schemas import ResourceDocumentSchema, ResourceSchema
 
 
 async def get_resource(request: web.Request) -> web.Response:
     """Endpoint to get a resource from the DB
     Respond with a 200 status code and a JSON body with the resource data
     If resource is not found, respond with a 404 status code
     """
-    [resource_id] = get_request_params(request, params_names=["resource_id"])
+
+    try:
+        resource_id = str(uuid.UUID(request.match_info["resource_id"]))
+    except Exception as e:
```

Comment:
It's a `ValueError` that's raised, right? Would we want to specify the kind of exception?

Reply:
It can be either a `ValueError`, either an error because it's not a valid UUID.
Could be indeed nice to go further and only catch those two specific exceptions.

Reply:
I'll keep that for a future PR though :)